src/error_types.rs

```diff
@@ -0,0 +1,33 @@
+use std::error;
+use std::fmt;
+use std::io;
+
+#[derive(Debug, Clone)]
+pub struct GnuplotInitError;
+
+impl fmt::Display for GnuplotInitError
+{
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result
+	{
+		write!(
+			f,
+			"Couldn't spawn gnuplot. Make sure it is installed and available in PATH."
+		)
+	}
+}
+
+impl error::Error for GnuplotInitError
+{
+	fn source(&self) -> Option<&(dyn error::Error + 'static)>
+	{
+		None
+	}
+}
+
+impl From<io::Error> for GnuplotInitError
+{
+    fn from(_error: io::Error) -> Self
+	{
+        GnuplotInitError
+    }
```

Comment:
Please run `cargo fmt` to fix some of these formatting errors.

Reply:
Ah, oops. I had had an issue running the formatter because it required rustfmt 1.2.0 instead of 1.3.0, which changed the names of a few flags. I wasn't able to figure out how to install an older rustfmt version, so I made a version of rustfmt.toml that's 1.3.0 compliant; would you like me to include that in the PR as well?

Reply:
Yes, I'd appreciate the update.